app/controllers/UpfrontPaymentController.scala

```diff
@@ -77,46 +81,77 @@ class UpfrontPaymentController @Inject() (
+      case Some(value) => value
+      case None        => Errors.throwBadRequestException("I am an error, this should never happen")
+    }
+    val totalDebtAmountInPence: AmountInPence = AmountInPence(debtTotalAmount.value)
+
+    UpfrontPaymentAmountForm.form(request.journey, minimumUpfrontPaymentAmount)
       .bindFromRequest()
       .fold(
-        formWithErrors =>
-          Future.successful(Ok(views.upfrontPaymentAmountPage(formWithErrors, mockJourney.qualifyingDebt, AmountInPence(100L)))),
-        (s: BigDecimal) => {
-          /* TODO: compute what is remaining to pay by subtracting "s" from the initial qualifying debt amount
-             and write to session store
-           */
-          Future(Redirect(routes.UpfrontPaymentController.upfrontSummary()))
+        (formWithErrors: Form[BigDecimal]) =>
+          Future.successful(Ok(
+            views.upfrontPaymentAmountPage(
+              form           = formWithErrors,
+              maximumPayment = totalDebtAmountInPence,
+              // todo, find out what the minimum upfront payment amount can be, for now £1
+              minimumPayment = minimumUpfrontPaymentAmount,
+              backUrl        = Some(routes.UpfrontPaymentController.canYouMakeAnUpfrontPayment().url)
+            )
+          )),
+        (validForm: BigDecimal) => {
+          //amount in pence case class apply method converts big decimal to pennies
+          val amountInPence: AmountInPence = AmountInPence(validForm)
+          journeyService.updateUpfrontPaymentAmount(request.journeyId, UpfrontPaymentAmount(amountInPence))
+            .map(_ => Redirect(routes.UpfrontPaymentController.upfrontPaymentSummary().url))
         }
       )
   }
 
-  val upfrontSummary: Action[AnyContent] = as.authenticatedJourneyAction.async { implicit request =>
-    val mockUserAnswers = UserAnswers.empty.copy(
-      hasUpfrontPayment = Some(true),
-      upfrontAmount     = Some(AmountInPence(10000L))
-    )
-    Future.successful(Ok(views.upfrontSummaryPage(mockUserAnswers, AmountInPence(200000L))))
+  val upfrontPaymentSummary: Action[AnyContent] = as.eligibleJourneyAction { implicit request =>
+    Ok("Submitted the upfront payment amount, page is still a wip and needs wiring up...\n" +
+      s"For testing purposes for now, here's the journey:\n ${Json.prettyPrint(Json.toJson(request.journey))}")
   }
 }
 
 object UpfrontPaymentController {
-
-  def upfrontPaymentAmountForm(journey: MockJourney): Form[BigDecimal] = Form(
-    mapping(
-      "UpfrontPaymentAmount" -> Forms.of(amountOfMoneyFormatter(AmountInPence(100L).inPounds > _, journey.qualifyingDebt.inPounds < _))
-    )(identity)(Some(_))
-  )
+  def determineMaxDebt(journey: Journey): Option[DebtTotalAmount] = journey match {
+    case _: Journey.BeforeEligibilityChecked => None
+    case j: Journey.AfterEligibilityChecked =>
+      Some(j.eligibilityCheckResult.chargeTypeAssessment.map(_.debtTotalAmount)
```

Comment:
this would make more sense as just
 ```scala
j.eligibilityCheckResult.chargeTypeAssessment.map(_.debtTotalAmount).headOption
```
The `Option` in the return type signifies the error case will be returned as a `None` rather than an exception being thrown. The place calling this is already handling the `None` case.

Reply:
just implemented this, resolving once I've pushed